rsyslog/go/src/github.com/mmexternal/undefined_field/undefined_field.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"encoding/json"
+	"bufio"
+	"strconv"
+	"strings"
+	"fmt"
+	"os"
+)
+
+const (
+	initial_use_undefined              = false
+	initial_undefined_name             = "undefined"
+	initial_undefined_to_string        = false
+	initial_default_keep_fields        = "CEE,time,@timestamp,aushape,ci_job,collectd,docker,fedora-ci,file,foreman,geoip,hostname,ipaddr4,ipaddr6,kubernetes,level,message,namespace_name,namespace_uuid,offset,openstack,ovirt,pid,pipeline_metadata,rsyslog,service,systemd,tags,testcase,tlog,viaq_msg_id"
+	initial_extra_keep_fields          = ""
+	initial_keep_empty_fields          = ""
+	initial_undefined_max_num_fields   = -1
+	initial_undefined_dot_replace_char = "UNUSED"
+	initial_logging_file_path          = "/var/log/rsyslog/rsyslog.log"
+)
+
+var (
+	use_undefined bool
+	keep_fields []string
+	keep_empty_fields []string
+	undefined_name string
+	undefined_to_string bool
+	undefined_dot_replace_char string
+	undefined_max_num_fields int64
+	logfile *os.File
+	noaction          = false
+	replacer          = &strings.Replacer{}
+	basic_keep_fields = "_STREAM_ID,_SYSTEMD_INVOCATION_ID"
```

Comment:
we really should add these to https://github.com/ViaQ/elasticsearch-templates/blob/master/namespaces/systemd.yml#L43 , then add them to https://github.com/openshift/cluster-logging-operator/blob/master/files/rsyslog/65-viaq-formatting.conf#L91 . . . but ok for now